openerp/fields.py

```diff
@@ -879,10 +879,19 @@ def _compute_value(self, records):
         for field in computed:
             records._cache[field] = field.null(records.env)
             records.env.computed[field].update(records._ids)
+        # if compute_sudo bypass access rights checks when computing values
+        targets = records.sudo() if self.compute_sudo else records
+        for record, target in zip(records, targets):
+            if not record.id and record.env != target.env:
+                # draft records: copy record's cache to target's cache first
+                copy_cache(record, target.env)
         if isinstance(self.compute, basestring):
-            getattr(records, self.compute)()
+            getattr(targets, self.compute)()
         else:
-            self.compute(records)
+            self.compute(targets)
+        if self.compute_sudo:
+            for record, target in zip(records, targets):
+                record[self.name] = target[field.name]
```

Comment:
@StefanRijnhart Good catch. I need to improve my fix since the same method can be used to compute a lot of field and the compute_sudo flag is by field. Therefore the method must be called with `sudo` for some fields and without `sudo` for others...